core/src/main/java/org/apache/iceberg/view/BaseMetastoreViewCatalog.java

```diff
@@ -67,6 +71,31 @@ public View loadView(TableIdentifier identifier) {
     throw new NoSuchViewException("Invalid view identifier: %s", identifier);
   }
 
+  @Override
+  public Table loadTable(TableIdentifier identifier) {
+    try {
+      return super.loadTable(identifier);
+    } catch (NoSuchTableException e) {
+      return loadViewMetadataTable(identifier);
```

Comment:
Similar to the other comment.  I don't feel this is the right way to handle metadata tables for views.  It should be handled as part of the metadata table fallback logic.